### PR TITLE
Improvement: DO-3223: Skip recalculating graph layout if size hasn't changed

### PR DESCRIPTION
**Pull request by Roman-Kornev**
## Motivation and Context

1. Layout calculation was done twice for the first render due to both the original `updateLayout` calculation and a `debouncedUpdateLayout` on `ResizeObserver`. The graph now stores its previous size and only recalculates the layout if the size actually changed.
This results in ~33% reduction in first frame render time.

2. A similar issue was found in the `useCausalGraphEditor` update effect. The `causalGraphParser` was called twice. 
Just reusing the result reduced the update time by another ~33%.

## How Has This Been Tested?

Modified the random cluster story to allow profiling first frame render times.

## PR Checklist:

- [x] I have implemented all requirements? (see JIRA, project documentation).
- [x] I am not affecting someone else's work, If I am, they are included as a reviewer.
- [x] I have added relevant tests (unit, integration or regression).
- [x] I have added comments to all the bits that are hard to follow.
- [x] I have added/updated Documentation.
- [x] I have updated the appropriate changelog with a line for my changes.

### First render:
BEFORE:
image
AFTER:
image

### On graph update:
BEFORE:
image
AFTER:
image

### CODE REVIEW

**Review by patricia-causalens — APPROVED** [review]
Nice find! Had a quick play on storybook and LGTM

**Review by krzysztof-causalens — APPROVED** [review]
Nice find!

**Review by tamara-causalens — APPROVED** [review]
Excellent find Roman!
